Automattic/woocommerce-payments
# Avoid registering express button hooks when payments are disabled

Fixes #5519

#### Changes proposed in this Pull Request

This PR ensures that express checkout buttons such as the WooPay, Apple Pay, and Google Pay buttons do not appear when payments are disabled on the WCPay account. 

#### Testing instructions

1. On the `develop` branch, enable Payment Requests and/or WooPay to ensure express checkout buttons will appear on the Checkout, Cart, or Product pages.
2. In the WooCommerce Payments Server codebase, open the accounts controller and change the [value of `payments_enabled`](https://github.com/Automattic/woocommerce-payments-server/blob/592789030a59f34f2bbc3bdc240112a3ded05866/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php#L668) to `false`.
3. Back on the WCPay client, clear the account cache via WCPay Dev Tools and double check the account data to ensure `payments_enabled` is set to `false`. 
4. Visit a Cart, Checkout, or Product page and notice that the express checkout/payment request buttons still appear.
6. Checkout the `fix/5519-disabled-account-hide-express-buttons` branch and refresh the page from above, verifying that the express checkout buttons do not appear.

-----


**Post merge**
